File: algorithms/2_medium/C/03895_count_digit_appearances/solution.py
class Solution:
    def countDigitOccurrences(self, nums: list[int], digit: int) -> int:
        freq = 0

        for num in nums:
            while num > 0:
                if num % 10 == digit:
                    freq += 1
                num //= 10

        return freq

File: algorithms/2_medium/C/03895_count_digit_appearances/solution.cpp
class Solution {
public:
    static int countDigitOccurrences(const vector<int>& nums, const int digit) {
        int freq  = 0;

        for (int num : nums) {
            while (num > 0) {
                if (num % 10 == digit) {
                    freq += 1;
                }
                num /= 10;
            }
        }

        return freq;
    }
};
